src/librustc/hir/lowering/expr.rs

```diff
@@ -508,15 +508,16 @@ impl LoweringContext<'_> {
     /// Desugar `<expr>.await` into:
     /// ```rust
     /// {
-    ///     let mut pinned = <expr>;
-    ///     loop {
-    ///         match ::std::future::poll_with_tls_context(unsafe {
-    ///             ::std::pin::Pin::new_unchecked(&mut pinned)
-    ///         }) {
-    ///             ::std::task::Poll::Ready(result) => break result,
-    ///             ::std::task::Poll::Pending => {},
+    ///     match future {
```

Comment:
For the sake of clarity the `future` on this line should probably be `<expr>`

Reply:
Fixed, thanks!